java/src/jmri/jmrix/lenz/XNetTurnout.java

```diff
@@ -526,19 +538,9 @@ protected synchronized void sendOffMessage() {
             log.debug("Current Thread ID: {} Thread Name {}", java.lang.Thread.currentThread().getId(), java.lang.Thread.currentThread().getName());
         }
         XNetMessage msg = getOffMessage();
-        // Set the known state to the commanded state.
-        // To avoid some of the command station busy
-        // messages, add a short delay before sending the
-        // first off message.
-            if (internalState != OFFSENT) {
-            jmri.util.ThreadingUtil.runOnLayoutDelayed( () ->
```

Comment:
This is interesting. On public mailing list, it was said:

> The series of off messages, including the delays, ensure that the command station tells the turnout to throw then actually stops sending the message to the rails, confirms the stop was received, and doesn’t tell us “I’m busy, go away for a while” in the process.
It was a hard fought painful process to get it to the point where it is and simplification of the process is not an option.

Now I see that a delay between ON message and a corresponding OFF goes away. It was also said that (some) delay is necessary 

It was also defined that
> > Q3: I understood that if OFF message was sent "immediately", in less than 30ms after the "ON" message, the command station/turnout decoder might not have enough time to even start the operation - the turnout might not be switched. Correct ?

> **Correct.**

Now it seems the OFF message enters the high priority queue immediately; where is that safety delay ensured  - or is it no longer necessary ?

Reply:
It isn’t ensured.

But you convinced me it may not be needed, because if we triggered two turnouts nearly simultaneously, the second off request might have actually been sent before the first one.

If we need to add it back in, we will, but it might not be in the same place.

This is part of the reason for putting this into the 4.21 series instead of the 4.19 series.  We can put it in the hands of users without creating a bug in the 4.20 production release.

Reply:
Well, that was never the intent - because your explanation (paraphrase) "to give the command station the time to REPEAT the instruction" on DCC is perfectly reasonable.

I never seen that happen, but IMHO ensure certain delay between ON and OFF is *very* reasonable - and I happen to know hardware designs which can easily miss a DCC packet occasionally, for example when its tiny CPU is doing something important, like timing signal for a servo. 

The intent was to bring other bugs to attention that cause one OFF to be scheduled, while other (duplicite) OFFs were sent right away without any delays by this particular code.

Reply:
On a second thought, there won't be any bug reports; because of the way how 'high priority' packets are processed for the last 10 years or so, they will be delayed after all pending messages are sent +100ms at least.